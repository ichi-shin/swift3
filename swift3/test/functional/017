#!/bin/bash

# List Multipart Uploads

. ./common

S3USER=tester

_s3_put /bucket

upload_id=$(_s3_post /bucket/object?uploads | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')
_s3_put /bucket/object?uploadId=${upload_id}\&partNumber=1 -T /dev/null

_s3_post /bucket/object?uploads | _filter_curl xml
_s3_post /bucket/object?uploads | _filter_curl xml
_s3_post /bucket/object?uploads | _filter_curl xml
_s3_post /bucket/object?uploads | _filter_curl xml

_s3_get /bucket?uploads\&encoding-type=url\&max-uploads=2 -D - | _filter_curl xml