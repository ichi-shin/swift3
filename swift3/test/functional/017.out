QA output created by 017
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X POST http://SWIFT_HOST/bucket/object?uploads... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=1... 200
> s3curl --id tester -- -X POST http://SWIFT_HOST/bucket/object?uploads... 200
<?xml version='1.0' encoding='UTF-8'?>
<InitiateMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Bucket>bucket</Bucket>
  <Key>object</Key>
  <UploadId>UPLOAD_ID</UploadId>
</InitiateMultipartUploadResult>
> s3curl --id tester -- -X POST http://SWIFT_HOST/bucket/object?uploads... 200
<?xml version='1.0' encoding='UTF-8'?>
<InitiateMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Bucket>bucket</Bucket>
  <Key>object</Key>
  <UploadId>UPLOAD_ID</UploadId>
</InitiateMultipartUploadResult>
> s3curl --id tester -- -X POST http://SWIFT_HOST/bucket/object?uploads... 200
<?xml version='1.0' encoding='UTF-8'?>
<InitiateMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Bucket>bucket</Bucket>
  <Key>object</Key>
  <UploadId>UPLOAD_ID</UploadId>
</InitiateMultipartUploadResult>
> s3curl --id tester -- -X POST http://SWIFT_HOST/bucket/object?uploads... 200
<?xml version='1.0' encoding='UTF-8'?>
<InitiateMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Bucket>bucket</Bucket>
  <Key>object</Key>
  <UploadId>UPLOAD_ID</UploadId>
</InitiateMultipartUploadResult>
> s3curl --id tester -- -X GET -D - http://SWIFT_HOST/bucket?uploads&encoding-type=url&max-uploads=2... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: application/xml
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<ListMultipartUploadsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Bucket>bucket</Bucket>
  <KeyMarker/>
  <UploadIdMarker/>
  <NextKeyMarker>object</NextKeyMarker>
  <NextUploadIdMarker>UPLOAD_ID</NextUploadIdMarker>
  <MaxUploads>2</MaxUploads>
  <EncodingType>url</EncodingType>
  <IsTruncated>true</IsTruncated>
  <Upload>
    <Key>object</Key>
    <UploadId>UPLOAD_ID</UploadId>
    <Initiator>
      <ID>TESTER</ID>
      <DisplayName>TESTER</DisplayName>
    </Initiator>
    <Owner>
      <ID>TESTER</ID>
      <DisplayName>TESTER</DisplayName>
    </Owner>
    <StorageClass>STANDARD</StorageClass>
    <Initiated>DATE</Initiated>
  </Upload>
  <Upload>
    <Key>object</Key>
    <UploadId>UPLOAD_ID</UploadId>
    <Initiator>
      <ID>TESTER</ID>
      <DisplayName>TESTER</DisplayName>
    </Initiator>
    <Owner>
      <ID>TESTER</ID>
      <DisplayName>TESTER</DisplayName>
    </Owner>
    <StorageClass>STANDARD</StorageClass>
    <Initiated>DATE</Initiated>
  </Upload>
</ListMultipartUploadsResult>
